framework/src/userobject/GeneralUserObject.C

```diff
@@ -39,6 +39,13 @@ GeneralUserObject::getSuppliedItems()
   return _supplied_vars;
 }
 
+void
+GeneralUserObject::addUserObjectDependencyHelper(const UserObject & uo) const
+{
+  UserObject::addUserObjectDependencyHelper(uo);
+  _depend_vars.insert(uo.name());
```

Comment:
We have `_depend_uos` in `UserObject`, I do not understand the difference between it and `_depend_vars`. But `_depend_vars` is not added here. `_depend_uos` has a complicated implementation in `getDependObjects` to resolve chain dependencies, which this `_depend_vars` may not have. I do not see why we need both. Who added this `_depend_vars`?

Reply:
@lindsayad pointed to me the change by me for introducing `_depend_uos`. Now I think the two `_depend_uos` and `_depend_vars` are serving different purpose. The change here is patching a hole in the system for `_depend_vars`. Why is this in `GeneralUserObject`? I am asking because I am wondering do we have issues for other types of user objects?

Reply:
Btw, the question is unrelated with this PR, so this conversation can be marked as resolved.